server/v2/store/go.mod

```diff
@@ -0,0 +1,72 @@
+module cosmossdk.io/server/v2/store
```

Comment:
why the new go.mod? can we use store/v2 as the home for this?

Reply:
Right now this new server module is just for saving prune cmd & maybe snapshot in the future.
@julienrbrt any idea?

Reply:
Also should we move it back to `cometbft` since we already have some snapshot helper there

Reply:
We were thinking about putting it in the `cometbft` server first indeed, but pruning isn't cometbft specific, so we need a better place for it.

We may not want to add a server/v2 and cobra dep in server/v2. An additional server did seem a good idea.

However, given that api and grpc haven't their own go.mod, maybe let's just keep the separate server component but kill the go.mod 👍🏾